git-archive-all.py

```diff
@@ -0,0 +1,505 @@
+        version="%prog {0}".format(__version__)
+    )
+
+    parser.add_option('--prefix',
+                      type='string',
+                      dest='prefix',
+                      default=None,
+                      help="""prepend PREFIX to each filename in the archive.
+                          OUTPUT_FILE name is used by default to avoid tarbomb.
+                          You can set it to '' in order to explicitly request tarbomb""")
+
+    parser.add_option('-v', '--verbose',
+                      action='store_true',
+                      dest='verbose',
+                      help='enable verbose mode')
+
+    parser.add_option('--no-exclude',
+                      action='store_false',
+                      dest='exclude',
+                      default=True,
+                      help="don't read .gitattributes files for patterns containing export-ignore attrib")
+
+    parser.add_option('--force-submodules',
+                      action='store_true',
+                      dest='force_sub',
+                      help='force a git submodule init && git submodule update at each level before iterating submodules')
+
+    parser.add_option('--extra',
+                      action='append',
+                      dest='extra',
+                      default=[],
+                      help="any additional files to include in the archive")
+
+    parser.add_option('--dry-run',
+                      action='store_true',
+                      dest='dry_run',
+                      help="don't actually archive anything, just show what would be done")
+
+    options, args = parser.parse_args()
+
+    if len(args) != 1:
+        parser.error("You must specify exactly one output file")
+
+    output_file_path = args[0]
+
+    if path.isdir(output_file_path):
+        parser.error("You cannot use directory as output")
+
+    # avoid tarbomb
+    if options.prefix is not None:
+        options.prefix = path.join(options.prefix, '')
+    else:
+        import re
+
+        output_name = path.basename(output_file_path)
+        output_name = re.sub(
+            '(\.zip|\.tar|\.tgz|\.txz|\.gz|\.bz2|\.xz|\.tar\.gz|\.tar\.bz2|\.tar\.xz)$',
+            '',
+            output_name
+        ) or "Archive"
+        options.prefix = path.join(output_name, '')
+
+    try:
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setFormatter(logging.Formatter('%(message)s'))
+        GitArchiver.LOG.addHandler(handler)
+        GitArchiver.LOG.setLevel(logging.DEBUG if options.verbose else logging.INFO)
+        archiver = GitArchiver(options.prefix,
+                               options.exclude,
+                               options.force_sub,
+                               options.extra)
+        archiver.create(output_file_path, options.dry_run)
+    except Exception as e:
+        parser.exit(2, "{0}\n".format(e))
+
+    sys.exit(0)
+
+
+if __name__ == '__main__':
+    main()
+
```

Comment:
Could you give a link to where this script is hosted/developed originally? That would be useful to update the script in the future or forward bug fixes if the need arises for either of these.